skimage/segmentation/random_walker_segmentation.py

```diff
@@ -505,6 +507,15 @@ def random_walker(
             'iterable with one number per spatial dimension.'
         )
 
```

Comment:
It might be worth changing line 498 (above) to: `np.ones(3, dtype=data.dtype)` 
So it doesn’t create a float64 array which forces the problem to be upcast to float64 in numpy 2.x.

I’m not sure what to do in the asarray case covered by line 503. Do we make the user ensure spacing and data have the right dtype? Or do we convert the spacing dtype to the data dtype?  When creating the problem later on, spacing and data will be divided and the result is thus the larger of the two dtypes for numpy 2.x.  And someone who provides a list array-like will get float64 for sure.

Reply:
Good catch! I'll give it a more thorough look again.

Reply:
I would convert the spacing dtype to the data dtype *and* issue a warning, so it's clear to the user without being disruptive. But this change could lead to slightly different end results... So, technically, this would take a deprecation cycle?

Reply:
See https://github.com/scikit-image/scikit-image/pull/7655/commits/ef1bc39646eae6703d2fa6de14a12c134e026397 and https://github.com/scikit-image/scikit-image/pull/7655#discussion_r1927174341. I'm a bit confused how to handle this properly.